packages/studioCMS/src/components/exports/Navigation.astro

```diff
@@ -0,0 +1,113 @@
+---
+import { getPageList, getSiteConfig, Avatar } from "studiocms:components";
+import { StudioCMSRoutes } from "studiocms-dashboard:routeMap";
+import { externalNav } from "virtual:studiocms/_nav"
+import "./navigation.css";
+
+type Props = {
+    topLevelLinkCount?: number;
+    hideAvatar?: boolean;
+}
+
+const { hideAvatar, topLevelLinkCount = 3 } = Astro.props;
+
+const config = await getSiteConfig();
+const pageList = await getPageList();
+
+const { title } = config;
+const { mainLinks: { baseSiteURL } } = StudioCMSRoutes;
+
+type LinkProps = {
+    text: string;
+    href: string;
+};
+
+const links: LinkProps[] = [];
+
+const navPagesList = pageList.filter(page => page.showOnNav === true && page.package === "@astrolicious/studiocms")
+    .sort((a, b) => Date.parse(b.publishedAt.toString()) - Date.parse(a.publishedAt.toString()));
+
+const indexPage = navPagesList.find(page => page.slug === "index");
+
+if (indexPage) {
+    links.push({
+        text: indexPage.title || "Home",
+        href: import.meta.env.BASE_URL,
+    })
+}
+
+const restOfPages = navPagesList.filter(page => page.slug !== "index");
+
+for (const page of restOfPages) {
+    links.push({
+        text: page.title,
+        href: import.meta.env.BASE_URL + page.slug,
+    })
+}
+
+const externalLinks = Array.from(externalNav.values())
+
+for (const link of externalLinks) {
+    links.push({
+        text: link.text,
+        href: import.meta.env.BASE_URL + link.slug,
```

Comment:
This unsafe, `import.meta.env.BASE_URL` is not guaranteed to have a tailing slash and `link.slug` could have a leading slash resulting in a double slash `//`